src/impl.ml

```diff
@@ -305,8 +305,8 @@ let sha1_update_cstruct ctx buffer =
   let buf = buffer.Cstruct.buffer in
   let buffer' : (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.Array1.t = Bigarray.Array1.sub buf ofs len in
   (* XXX: need a better way to do this *)
-  let buffer'': (int,  Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.Array1.t = Obj.magic buffer' in
-  Sha1.update_buffer ctx buffer''
+  (* let buffer'': (int,  Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.Array1.t = Obj.magic buffer' in *)
```

Comment:
One less `Obj.magic` :) Still one remaining though when converting int to file descriptor.